qt/src/hocr/HOCRPdfExporter.cc

```diff
@@ -680,7 +732,45 @@ bool HOCRPdfExporter::setSource(const QString& sourceFile, int page, int dpi, do
 	}
 }
 
-QImage HOCRPdfExporter::getSelection(const QRect& bbox)
-{
+QImage HOCRPdfExporter::getSelection(const QRect& bbox) {
 	return m_displayerTool->getSelection(bbox);
 }
+
+void HOCRPdfExporter::paperSizeChanged() {
+	PaperSizeMetrics currentMetric = static_cast<PaperSizeMetrics>(ui.comboBoxPaperSizeMetric->itemData
+	                                 (ui.comboBoxPaperSizeMetric->currentIndex()).toInt());
+
+	auto textPaperSize = ui.comboBoxPaperSize->itemData(ui.comboBoxPaperSize->currentIndex()).toString();
+	if(textPaperSize == "custom") {
+		ui.labelPaperSizeHeight->setVisible(true);
+		ui.labelPaperSizeWidth->setVisible(true);
+		ui.lineEditPaperSizeWidth->setVisible(true);
+		ui.lineEditPaperSizeHeight->setVisible(true);
+		ui.lineEditPaperSizeWidth->setEnabled(true);
+		ui.lineEditPaperSizeHeight->setEnabled(true);
+	} else if(textPaperSize == "source") {
+		ui.labelPaperSizeHeight->setVisible(false);
+		ui.labelPaperSizeWidth->setVisible(false);
+		ui.lineEditPaperSizeWidth->setVisible(false);
+		ui.lineEditPaperSizeHeight->setVisible(false);
+	} else {
+		ui.labelPaperSizeHeight->setVisible(true);
+		ui.labelPaperSizeWidth->setVisible(true);
+		ui.lineEditPaperSizeWidth->setVisible(true);
+		ui.lineEditPaperSizeHeight->setVisible(true);
+		ui.lineEditPaperSizeWidth->setEnabled(false);
+		ui.lineEditPaperSizeHeight->setEnabled(false);
```

Comment:
You could pack all these widgets in a group (in the ui designer), then you can all these lines to just
   
    ui.groupCustomPaperSize.setEnabled(textPaperSize == "custom");
    ui.groupCustomPaperSize.setVisible(textPaperSize != "source");

Reply:
I am not sure, is it required o r not. But i found that comboBoxPaperSizeMetric also should be invisible in "source" mode.

Reply:
Yep I'd agree that is should not be visible in source mode.

Reply:
Not sure about landscape checkbox - maybe it also should be invisible in source mode.

Reply:
yes agree

Reply:
Done.

Reply:
Because i am not sure, how i should group these widgets. And i wouldn't create difficult group hierarchy.

Reply:
just use a widget with a gridlayout and put them all inside this widget

Reply:
All - which of them? 
`ui.checkBoxLandscape
ui.comboBoxPaperSizeMetric
ui.labelPaperSizeHeight
ui.labelPaperSizeWidth
ui.lineEditPaperSizeWidth
ui.lineEditPaperSizeHeight
ui.lineEditPaperSizeWidth
ui.lineEditPaperSizeHeight`

Reply:
All except `ui.checkBoxLandscape` (and actually you could use a combobox [Portrait, Landscape] for the orientation, which is more in line with what all other applications do. I'd lay out as follows:
[ComboPaperSize][ComboPaperOrientation]
[Group Width/Height/Units]